### PR TITLE
Add `details`

### PR DESCRIPTION
**Pull request by genericallyterrible**
Adds new function `details` to automatically create a well formed [Details disclosure element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details) (`<details>`).

Currently scoped as:

~~using PlutoUI.ExperimentalLayout: details~~

```julia
using PlutoUI #: details
```

Example usage of new details function

Since this new function creates normal HTML `<details>` elements, collapsed details elements will omit their contents from any print/export. If this behavior is undesirable, it seems a JS fix could be applied as discussed [here on SO](https://stackoverflow.com/questions/19646684/force-open-the-details-summary-tag-for-print-in-chrome#answer-75260733) though I'm unsure where it would slot in.

I couldn't get `@bind` to behave properly in the Layout.jl notebook to fully test it internally. UI elements would render, but bound variables would never update. When using `details` in the binder environment automatically created by github-actions it behaves as expected.

Example usage of new details function with bound variables

### CODE REVIEW

File: src/Layout.jl
Comment (fonsp) [inline src/Layout.jl:633]:
```suggestion
				$(Iterators.map(contents) do detail
```

**Comment by fonsp** [comment]
Super nice!! And really well implemented, happy to see that you are using HypertextLiteral!! I would say that this is not "experimental", and it can go directly into PlutoUI. What do you think?

**Comment by genericallyterrible** [comment]
Sounds good to me! Wasn't super sure where `details` should end up, and honestly stumbled my way into Layout.jl. If it should live in some other file so it's not submoduled under ExperimentalLayout, the only reliance it has on the other code in Layout.jl is the 5 lines implementing `embed_display` (and the variable `p` for example purposes).

I can move `details` into its own notebook and copy over the functionality of `embed_display` if that's appropriate, or perhaps there's some other path you had in mind?

**Comment by genericallyterrible** [comment]
`details` has been moved to Details.jl and included directly into PlutoUI.jl

File: src/Details.jl
Comment (fonsp) [inline src/Details.jl:80]: FYI this can later be replaced with https://github.com/JuliaPluto/AbstractPlutoDingetjes.jl/pull/9

File: src/Details.jl
Comment (fonsp) [inline src/Details.jl:91]: I decided to drop the extra plutoui-details element, and wrap an AbstractString inside `<p>`. It looks good to me now, but could you check whether this is still the look that you intended?

**Review by fonsp — APPROVED** [review]
Hey! Again this PR is really impressive! Thanks so much for the contribution :)

After this PR is merged, I would like to move all of the CSS into Pluto itself: a details element created without PlutoUI should also get these nice styles :) Can you make a PR to Pluto for that?

I made some small tweaks, see the comments. To me this is ready to merge, let me know!

**Comment by genericallyterrible** [comment]
Thanks so much for the tweaks and tests!

I'd be happy to PR the CSS over to Pluto once this is accepted, everyone deserves pretty `<details>`. 😊

I have a bit of clarification on one of your changes and its impacts on the assumptions I made about how `details` should format its contents, but I'll leave that comment as a response to the change itself.

File: src/Details.jl
Comment (genericallyterrible) [reply src/Details.jl:91]: I love how this change would really simplify the code (and CSS!), yet it strikes at a core question: "What are the default assumptions and guarantees of the `details` function?" I should have expanded briefly on my philosophy and assumptions for `details` at the start of this PR so it could quickly be checked for alignment with the rest of the project, a lesson I will carry forward with me.

## Gotta wrap 'em all?

One goal of my initial design was to ensure consistent spacing between the items of the `content` argument as if they were all `<p>` elements under the assumption that this would be the expected behavior in the simple case and that advanced users could easily format their content items ahead of time with other well known methods (`md_str`, `combine`, etc) if necessary. I'm in favor of maintaining this assumption, but in the end it's your call.

With this change currently, Details.jl actually contains a convenient example of behavior I'd consider undesirable.

| Recording of Details.jl dictionary toggle |
|:---:|
| _Demo of how some objects embedded by Pluto can now display unexpected behavior._ |

To fully demonstrate the impact I whipped up a mock notebook to make A/B comparison straightforward.

| Recording of alternative wrapping behavior demo |
|:--:|
| _Demo of the differences between wrapping all details and only wrapping `AbstractString` details._ |

## I just wanted to use `<p>` ;-;

Assuming we do want to continue wrapping all elements, `<p>` elements are unfortunately not valid when nested. A `md_str` can easily put us in a situation where `@htl` is forced to normalize the emitted DOM structure to something valid, leaving empty `<p>` tags on the final output. Visually, I see no difference, but it feels a little sloppy. I'm happy to pivot to a `<div>` with a class if that's favorable to the custom `<plutoui-detail>` element.

| Recording of alternative tag wrapping demo |
|:--:|
| _Demo of the differences between wrapping all details in `<plutoui-detail>` and wrapping all details in `<p>`._ |

**Comment by fonsp** [comment]
Thanks so much for the PR! Let me know what you would like to work on next :)

**Pull request merged**
